Add documentation for timescaledb.bgw_log_level (#2744) (#2771)
timescale/docs: use-timescale/troubleshoot-timescaledb.md

and then inspect `dump_file.txt` before sending it together with a bug report or support question.

## Debugging background jobs

By default, background workers do not print a lot of information about
execution. The reason for this is to avoid writing a lot of debug
information to the PostgreSQL log unless necessary.

To aid in debugging the background jobs, it is possible to increase
the log level of the background workers without having to restart the
server by setting the `timescaledb.bgw_log_level` GUC and reloading
the configuration.

```sql

> ALTER SYSTEM SET timescaledb.bgw_log_level TO 'DEBUG1';

Author: ```suggestion
ALTER DATABASE SET timescaledb.bgw_log_level TO 'DEBUG1';
```

| Custom job        | Execution started                                    |
| Recompression job | Recompression job completed                          |
| Reorder job       | Chunk reorder completed                              |
| Reorder job       | Chunk reorder started                                |
| Scheduler         | New jobs discovered and added to scheduled jobs list |
| Scheduler         | Scheduling job for launch                            |

### Debug level 2

The amount of information printed at each level varies between jobs,
but the information printed at `DEBUG2` is currently shown below.

Note that all messages at level `DEBUG1` are also printed when you set
the log level to `DEBUG2`, which is [normal PostgreSQL
behaviour][log_min_messages].

| Source    | Event                              |
|-----------|------------------------------------|
| All jobs  | Job found in jobs table            |
| All jobs  | Job starting execution             |
| Scheduler | Scheduled jobs list update started |
| Scheduler | Scheduler dispatching job |

### Debug level 5

| Source    | Event                                |
|-----------|--------------------------------------|

> | Scheduler | Scheduled wakeup                     |

Author: ```suggestion
| Scheduler | Scheduled wake up                     |
```

and then inspect `dump_file.txt` before sending it together with a bug report or support question.

## Debugging background jobs

By default, background workers do not print a lot of information about
execution. The reason for this is to avoid writing a lot of debug
information to the PostgreSQL log unless necessary.

To aid in debugging the background jobs, it is possible to increase
the log level of the background workers without having to restart the
server by setting the `timescaledb.bgw_log_level` GUC and reloading
the configuration.

```sql
ALTER SYSTEM SET timescaledb.bgw_log_level TO 'DEBUG1';
SELECT pg_reload_conf();
```

This variable is set to the value of
[`log_min_messages`][log_min_messages] by default, which typically is
`WARNING`. If the value of [`log_min_messages`][log_min_messages] is
changed in the configuration file, it is used for
`timescaledb.bgw_log_level` when starting the workers.

> **NOTE:** Note that both `ALTER SYSTEM` and `pg_reload_conf()`

> > requires superuser privileges by default and that you need to grant
> > `EXECUTE` permissions to `pg_reload_conf()` and `ALTER SYSTEM`
> > privileges to `timescaledb.bgw_log_level` if you want this to work
> > for a non-superuser. Since `ALTER SYSTEM` privileges only exist on
> > PostgreSQL 15 and later, the necessary grants for executing these
> > statements just exists on TimescaleDB Cloud for PostgreSQL 15 or
> > later.

Reviewer: ```suggestion
> requires superuser privileges by default and that you need to grant
> `EXECUTE` permissions on `pg_reload_conf()` and `ALTER SYSTEM`
> privileges on `timescaledb.bgw_log_level` if you want this to work
> for a non-superuser. Since `ALTER SYSTEM` privileges only exist on
> PostgreSQL 15 and later versions, you need at least PostgreSQL 15
> to use this parameter on Timescale Cloud.
```
